routing/pathfind.go

```diff
@@ -590,11 +591,13 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 			timeLockDelta = edge.TimeLockDelta
 		}
 
-		incomingCltv := toNodeDist.incomingCltv +
-			uint32(timeLockDelta)
+		incomingCltv := toNodeDist.incomingCltv + int32(timeLockDelta)
+
+		// Check that we are within our CLTV limit. Take care with this
+		// comparison, because the cltv limit may be MaxUint32.
+		if uint64(incomingCltv) >
+			uint64(r.CltvLimit)+uint64(finalHtlcExpiry) {
```

Comment:
can this line be extracted sooner, and calculated once?

Reply:
extracted